lib/compress.js

```diff
@@ -2187,15 +2188,16 @@ merge(Compressor.prototype, {
 
     // determine if expression is constant
     (function(def){
-        function all(list) {
+        function all(compressor, list) {
             for (var i = list.length; --i >= 0;)
-                if (!list[i].is_constant_expression())
+                if (!list[i].is_constant_expression(compressor))
                     return false;
             return true;
         }
         def(AST_Node, return_false);
         def(AST_Constant, return_true);
-        def(AST_Lambda, function(scope){
+        def(AST_Lambda, function(compressor, scope){
+            if (!compressor.option("reduce_funcs")) return false;
```

Comment:
@alexlamsl This condition could be more specific - only look at AST_Defun, perhaps allow cases where constants are returned.